src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/TypeNamesShouldNotMatchNamespaces.cs

```diff
@@ -158,9 +158,7 @@ private static void InitializeWellKnownSystemNamespaceTable()
             {
                 lock (s_lock)
                 {
-#pragma warning disable CA1508 // Avoid dead conditional code - https://github.com/dotnet/roslyn-analyzers/issues/3861
                     if (s_wellKnownSystemNamespaceTable == null)
-#pragma warning restore CA1508 // Avoid dead conditional code
                     {
```

Comment:
It looks like the `lock` false positive was fixed 4 years ago in https://github.com/dotnet/roslyn-analyzers/commit/cd4302c1f7499514f795269f1e819b2382194b07

The reported issue is much newer than the fix.